crates/ruff_python_parser/src/lexer.rs

```diff
@@ -621,13 +621,22 @@ impl<'src> Lexer<'src> {
                 self.cursor.bump();
                 quote
             }),
-            (_, second) if is_quote(self.cursor.second()) => {
-                self.try_double_char_prefix([first, second]).then(|| {
+            (_, second) if is_quote(self.cursor.second()) => self
+                .try_double_char_prefix([first, second])
+                .and_then(|prefix_kind| {
+                    if matches!(prefix_kind, DoubleCharPrefix::Incompatible) {
+                        // This produces an `Unknown` token but we do not
+                        // need to record it.
+                        self.push_error(LexicalError::new(
```

Comment:
It would be helpful to include an explaination why it's okay to drop the unknown token (`push_token` should probably be marked with `#[must_use]`).

Is there a risk that any downstream user of the tokens or AST will drip over the fact that the token/ast prefix doesn't match the source prefix?